src/ontology/uberon.Makefile

```diff
@@ -1132,39 +1141,51 @@ collected-metazoan.owl: $(TMPDIR)/collected-metazoan.owl
 	$(ROBOT) merge -i $< $(COMPOSITE_STRIPPING_COMMAND) \
 		 annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $@
 
+# Step 1c: collected-lifestages is special in that it should not
+# include Uberon and CL (only the life-stages-minimal subset).
+$(TMPDIR)/collected-lifestages.owl: $(COLLECTED_lifestages_SOURCES)
+	$(ROBOT) merge $(foreach src,$^,-i $(src)) -o $@
+
+# Step 1d: And it is also a released artefact.
+collected-lifestages.owl: $(TMPDIR)/collected-lifestages.owl
+	$(ROBOT) merge -i $< $(COMPOSITE_STRIPPING_COMMAND) \
+		 annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $@
+
 # Step 2: Create a "composite" ontology. This is the core of the
 # composite pipeline. It heavily relies on the Uberon plugin for ROBOT,
 # which provides the 'merge-species' and 'merge-equivalent-sets'
 # commands.
-TAXON_GCI_RELS = RO:0002202 RO:0002496 RO:0002497 BFO:0000051
-MERGESPECIES_OPTS = --remove-declarations --extended-translation --translate-gcas
 .PRECIOUS: $(TMPDIR)/composite-%.owl
-$(TMPDIR)/composite-%.owl: $(TMPDIR)/collected-%.owl | all_robot_plugins
+$(TMPDIR)/composite-%.owl: $(TMPDIR)/collected-%.owl $(TMPDIR)/tax-merges.tsv | all_robot_plugins
 	$(ROBOT) merge -i $< $(COMPOSITE_STRIPPING_COMMAND) \
-		 uberon:merge-species $(MERGESPECIES_OPTS) -s 'mouse'      -t NCBITaxon:10090 $(foreach rel,$(TAXON_GCI_RELS),-q $(rel)) \
-		 uberon:merge-species $(MERGESPECIES_OPTS) -s 'human'      -t NCBITaxon:9606  $(foreach rel,$(TAXON_GCI_RELS),-q $(rel)) \
-		 uberon:merge-species $(MERGESPECIES_OPTS) -s 'primate'    -t NCBITaxon:9443 \
-		 uberon:merge-species $(MERGESPECIES_OPTS) -s 'Xenopus'    -t NCBITaxon:8353 \
-		 uberon:merge-species $(MERGESPECIES_OPTS) -s 'Danio'      -t NCBITaxon:7954 \
-		 uberon:merge-species $(MERGESPECIES_OPTS) -s 'Drosophila' -t NCBITaxon:7227 \
-		 uberon:merge-species $(MERGESPECIES_OPTS) -s 'C elegans'  -t NCBITaxon:6237 \
+		 uberon:merge-species --remove-declarations \
+		                      --extended-translation \
+		                      --translate-gcas \
+		                      --batch-file $(TMPDIR)/tax-merges.tsv \
 		 uberon:merge-equivalent-sets -s UBERON=10 -s CL=9 -s CARO=5 \
 		                              -l UBERON=10 -l CL=9 \
 		                              -d UBERON=10 -d CL=9 \
 		 reason -r ELK --equivalent-classes-allowed all \
 		 relax \
 		 reduce -r ELK -o $@
 
+# Step 2a: The "tax-merges.tsv" file used in the rule above is automatically
+# derived from the list of species in config/taxa.yaml.
+$(TMPDIR)/tax-merges.tsv: $(SCRIPTSDIR)/taxa.py config/taxa.yaml
+	python3 $(SCRIPTSDIR)/taxa.py make-merge-table > $@
```

Comment:
Optional: This is not a breaking comment but I have made the experience that to be able to read makefiles unobtrusively, it is nice if all file dependencies are passed explicitly to the python script. Else, oftencase, you have like dozens of file dependencies in the header of a python script which violates SRP when the file moves location, and also when a goal contains multiple processes (its then unclear which make dependency relates to which script).

Reply:
So you’d prefer something like

```
python3 $(SCRIPTSDIR)/taxa.py make-merge-table config/taxa.yaml > $@
```

?

No objection to that.